src/inference/include/openvino/runtime/properties.hpp

```diff
@@ -0,0 +1,663 @@
+    }
+
+    /**
+     * @brief compares property name
+     * @return true if string is the same
+     */
+    friend bool operator==(const std::string& str, const BaseProperty<T, mutability_>& property) {
+        return property == str;
+    }
+
+private:
+    const char* _name = nullptr;
+};
+template <typename T, PropertyMutability M>
+inline std::ostream& operator<<(std::ostream& os, const BaseProperty<T, M>& property) {
+    return os << property.name();
+}
+}  // namespace util
+/** @endcond */
+
+/**
+ * @brief This class is used to bind property name with value type
+ * @tparam T type of value used to set or get property
+ */
+template <typename T, PropertyMutability mutability_ = PropertyMutability::RW>
+struct Property : public util::BaseProperty<T, mutability_> {
+    using util::BaseProperty<T, mutability_>::BaseProperty;
+    /**
+     * @brief Constructs property
+     * @tparam Args property constructor arguments types
+     * @param args property constructor arguments
+     * @return Pair of name and type erased value.
+     */
+    template <typename... Args>
+    inline std::pair<std::string, Any> operator()(Args&&... args) const {
+        return {this->name(), Any::make<T>(std::forward<Args>(args)...)};
+    }
+};
+
+/**
+ * @brief This class is used to bind read-only property name with value type
+ * @tparam T type of value used to pass or get property
+ */
+template <typename T>
+struct Property<T, PropertyMutability::RO> : public util::BaseProperty<T, PropertyMutability::RO> {
+    using util::BaseProperty<T, PropertyMutability::RO>::BaseProperty;
+};
+
+/**
+ * @brief Read-only property to get a std::vector<PropertyName> of supported read-only properies.
+ *
+ * This can be used as a compiled model property as well.
+ *
+ */
+static constexpr Property<std::vector<PropertyName>, PropertyMutability::RO> supported_properties{
+    "SUPPORTED_PROPERTIES"};
+
+/**
+ * @brief Read-only property to get a std::vector<std::string> of available device IDs
+ */
+static constexpr Property<std::vector<std::string>, PropertyMutability::RO> available_devices{"AVAILABLE_DEVICES"};
+
+/**
+ * @brief Read-only property which defines support of import/export functionality by plugin
+ */
+static constexpr Property<bool, PropertyMutability::RO> import_export_support{"IMPORT_EXPORT_SUPPORT"};
+
+/**
+ * @brief Read-only property to get a name of name of a model
+ */
+static constexpr Property<std::string, PropertyMutability::RO> model_name{"NETWORK_NAME"};
+
+/**
+ * @brief Read-only property to get an unsigned integer value of optimal number of compiled model infer requests.
+ */
+static constexpr Property<uint32_t, PropertyMutability::RO> optimal_number_of_infer_requests{
+    "OPTIMAL_NUMBER_OF_INFER_REQUESTS"};
+
+namespace hint {
+
+/**
+ * @brief Enforce device to use specified precision for inference
+ */
+static constexpr Property<element::Type, PropertyMutability::RW> inference_precision{"INFERENCE_PRECISION_HINT"};
```

Comment:
@vladimir-paramuzov are you OK with such name? I suppose in your slides it was called something like `RUNTIME_PRECISION`

Reply:
confirmed that:
1. @dmitry-gorokhov are OK to support this property on CPU
2. @vladimir-paramuzov ok with the name